source/php/Controller/Project.php

```diff
@@ -42,6 +42,17 @@ public function singleViewController($data)
         );
 
         $data['scrollSpyMenuItems'] = $this->scrollSpyMenuItems($data);
+        $data['list'] = [];
+
+        if($data['project']['gallery']) {
```

Comment:
1. Lift up into an items of `$data['project']` above like meta etc.
2. Ensure we get either the gallery postMeta or an empty array
`$data['project']['gallery'] = WP::getPostMeta('gallery', [])` 
3. Include UI class with use (in controller):
`use ProjectManagerIntegration\UI\Gallery`
4. Refactor controller data using a postMetaData to viewData adapter in controller:
`$data['project']['gallery'] = Gallery::create(WP::getPostMeta('gallery', []))`
4. Add new class: ProjectManagerIntegration\UI\Gallery, with a static create method that always return an array (empty or with list)